packages/core/src/utils.js

```diff
@@ -634,6 +634,9 @@ export function stubExistingAdditionalProperties(
     properties: { ...schema.properties },
   };
 
+  // handle null formData
+  formData = formData || {};
```

Comment:
Shouldn't `formData = {}` already fix this on line 629? Why doesn't that work?

Reply:
I'm trying to understand when `formData` would ever be `null`? Won't it always be an object or `undefined`?

Reply:
Hi @epicfaace - thanks for looking into this! 

I bumped into this where I had a nested object property that is explicitly set to null (not undefined) which caused the error you will get if you comment my fix and run the provided test..

The actual code in the npm package for `stubExistingAdditionalProperties` contains

```javascript
var formData = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
```

which you can see doesn't handle null - only undefined. I think better would be 

```javascript
var formData = arguments.length > 2 && arguments[2] ? arguments[2] : {};
```

since null evaluates to false - but I don't know where this "transpilation" is coming from!? 

Happy to provide more context/info if needed!

Reply:
The transpilation happens because the original code says

```
export function stubExistingAdditionalProperties(
  schema,
  rootSchema = {},
  formData = {}
) 
```

Specifying a default argument of `{}` means that the default will be reverted to only if `formData` is undefined (but not if it is null).



> I bumped into this where I had a nested object property that is explicitly set to `null` (not `undefined`) which caused the error you will get if you comment my fix and run the provided test..

Interesting! Can you add a test that explicitly checks for this condition from the end user's point of view? i.e. a test in Form_test.js that sets the formData according to what you mentioned

Reply:
Hi @epicfaace - sorry for the delay - i've added the corresponding Form test - if you uncomment my fix you should get something like the following error when running `npm run test` 
```
@rjsf/core:   5) utils
@rjsf/core:        retrieveSchema()
@rjsf/core:          should handle null formData for schema which contains additionalProperties:
@rjsf/core:      TypeError: Cannot convert undefined or null to object
@rjsf/core:       at Function.keys (<anonymous>)
@rjsf/core:       at stubExistingAdditionalProperties (src/utils.js:674:10)
@rjsf/core:       at retrieveSchema (src/utils.js:758:12)
```

Hope this helps!

Reply:
Thanks for the test, I get what you're doing now. I think we want to handle not only null formData, but any non-object formData.

How about we do this instead:

```suggestion
formData = isObject(formData) ? formData : {};
```

And then let's also add an additional test where the formData is set to a non-null non-object (for example, when formData is equal to `2` or an array)?